libsolidity/ast/AST.cpp

```diff
@@ -44,6 +44,18 @@ ASTNode::ASTNode(int64_t _id, SourceLocation _location):
 {
 }
 
+Declaration const* ASTNode::referencedDeclaration(Expression const& _expression)
+{
+	if (auto const* memberAccess = dynamic_cast<MemberAccess const*>(&_expression))
+		return memberAccess->annotation().referencedDeclaration;
+	else if (auto const* identifierPath = dynamic_cast<IdentifierPath const*>(&_expression))
```

Comment:
Maybe it's better to skip this part, since currently, the two use cases is not expecting this.

Reply:
Yeah. Maybe it would be better to have separate `referencedVariableDeclaration()` and `referencedFunctionDeclaration()` that also cast the result to the appropriate declaration type? This would remove the need to include `IdentifierPath` for consistency.

Reply:
Oh, actually it does not help if `identifierPath` can be associated with a variable or a function.

Anyway, an assertion at the point of the call would probably be the best way to guard against it (if it's actually undesirable which depends on the calling code).